RuddockWebsite/validation_utils.py

```diff
@@ -0,0 +1,124 @@
+    if result is not None:
+      error = 'Username is already in use!'
+
+  if error is not None:
+    if flash_errors:
+      flash(error)
+    return False
+  return True
+
+def validate_password(password, password2, flash_errors=True):
+  '''
+  Checks to make sure a password is valid. password and password2 should be the
+  values from both times the user enters a password. If you need to check a
+  password outside of the context of a user setting/changing a password, then
+  use the same value for both inputs. If flash_errors is set to True, then the
+  errors will be displayed as flashes.
+
+  Returns True if valid, else False.
+  '''
+  error = None
+  if len(password) == 0:
+    error = 'You must provide a password!'
+  elif len(password2) == 0:
+    error = 'You must confirm your password!'
+  elif password != password2:
+    error = 'Passwords do not match. Please try again!'
+  elif len(password) < constants.MIN_PASSWORD_LENGTH:
+    error = 'Your password must be at least {0} characters long!'.format(
+        constants.MIN_PASSWORD_LENGTH)
+  elif len(password) > constants.MAX_PASSWORD_LENGTH:
+    error = 'Your password cannot be more than {0} characters long!'.format(
+        constants.MAX_PASSWORD_LENGTH)
+
+  if error is not None:
+    if flash_errors:
+      flash(error)
+    return False
+  return True
+
+def validate_name(name, flash_errors=True):
+  ''' Validates a name. Flashes errors if flash_errors is True. '''
+  # Allow letters, spaces, hyphens, and apostrophes.
+  # First and last characters must be letters.
+  name_regex = re.compile(r"^[a-z][a-z '-]{0,30}[a-z]$", re.I)
+  if not name_regex.match(name):
+    if flash_errors:
+      flash("'{0}' is not a valid name.".format(name))
+    return False
+  return True
+
+def validate_uid(uid, flash_errors=True):
+  ''' Validates a UID. Flashes errors if flash_errors is True. '''
+  uid_regex = re.compile(r'^[0-9]{7}$')
+  if not uid_regex.match(uid):
+    if flash_errors:
+      flash("'{0}' is not a valid UID.".format(uid))
+    return False
+  return True
+
+def check_uid_exists(uid):
+  ''' Returns True if the UID exists in the database. '''
+  query = text("SELECT 1 FROM members WHERE uid = :uid")
+  result = g.db.execute(query, uid=uid).first()
+  if result is None:
+    return False
+  return True
+
+def validate_year(year, flash_errors=True):
+  ''' Validates a year. Flashes errors if flash_errors is True. '''
+  year_regex = re.compile(r'^[0-9]{4}$')
+  # Also check that the year is in the range supported by MySQL's year type.
+  if year_regex.match(year) \
+      and int(year) >= 1901 \
+      and int(year) <= 2155:
+    return True
+  if flash_errors:
+    flash("'{0}' is not a valid year.".format(year))
+  return False
+
+def validate_email(email, flash_errors=True):
+  ''' Validates an email. Flashes errrors if flash_errors is True. '''
+  email_regex = re.compile(r'^[a-z0-9\.\_\%\+\-]+@[a-z0-9\.\-]+\.[a-z]{2,4}$', re.I)
```

Comment:
You shouldn't need to escape `._%+` inside the character classes. `-` (hyphen) can go at the end (or the beginning) without being escaped.